src/index.mjs

```diff
@@ -150,14 +199,12 @@ export function listen(options) {
 * @return {Object} a Promise
 */
 export function prefetch(url, isPriority, conn) {
-  if (conn = navigator.connection) {
-    // Don't prefetch if using 2G or if Save-Data is enabled.
-    if (conn.saveData) {
-      return Promise.reject(new Error('Cannot prefetch, Save-Data is enabled'));
-    }
-    if (/2g/.test(conn.effectiveType)) {
-      return Promise.reject(new Error('Cannot prefetch, network conditions are poor'));
-    }
+  checkConnection(conn = navigator.connection).catch(err => {
+    return Promise.reject(new Error('Cannot prefetch, '+err.message));
+  });
+  
+  if(toPrerender.size > 0 && !shouldPrerenderAndPrefetch){
```

Comment:
Throughout the PR, I noticed there is inconsistent application of spacing when compared to the code being replaced. In this block for example this should be:

```
if (toPrerender.size > 0 && !shouldPrerenderAndPrefetch) {
```

Could you do a style pass just to make sure conventions used elsewhere in the repo are being followed?

Reply:
Thanks for catching this Addy, took another pass and fixed conventions inconsistencies.